kousa/lib/gen/room_chat.ex

```diff
@@ -59,13 +59,14 @@ defmodule Kousa.Gen.RoomChat do
     {:noreply, state}
   end
 
-  def handle_cast({:new_msg, user_id, msg}, %State{} = state) do
+  def handle_cast({:new_msg, user_id, msg, whispered_to}, %State{} = state) do
     last_timestamp = Map.get(state.last_message_map, user_id)
     {_, seconds, _} = :os.timestamp()
 
     if not Map.has_key?(state.ban_map, user_id) and
-         (is_nil(last_timestamp) or seconds - last_timestamp > 0) do
-      ws_fan(state.users, :chat, %{
+      (is_nil(last_timestamp) or seconds - last_timestamp > 0) do
+      users = if whispered_to != [], do: Enum.filter(state.users, fn uid -> Enum.member?([user_id] ++ whispered_to, uid) end), else: state.users
```

Comment:
better way to add element to array: `[user_id | whispered_to]`

Reply:
also do it before you iterate

Reply:
whispered_to is an array itself

Reply:
Oh Okay got it @benawad